interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java

```
@@ -0,0 +1,574 @@
+              + "\n  --test_duration-secs=SECONDS   '-1' for no limit. Default: " + c.durationSecs
+              + "\n  --num_channels_per_server=INT  Number of connections to each server address."
+              + " Default: " + c.channelsPerServer
+              + "\n  --num_stubs_per_channel=INT    Default: " + c.stubsPerChannel
+              + "\n  --metrics_port=PORT            Listening port of the metrics server."
+              + " Default: " + c.metricsPort
+      );
+      System.exit(1);
+    }
+  }
+
+  @VisibleForTesting
+  void startMetricsService() throws IOException {
+    Preconditions.checkState(!shutdown, "client was shutdown.");
+
+    metricsServer = ServerBuilder.forPort(metricsPort)
+        .addService(MetricsServiceGrpc.bindService(new MetricsServiceImpl()))
+        .build()
+        .start();
+  }
+
+  @VisibleForTesting
+  void runStressTest() throws ExecutionException, InterruptedException {
+    Preconditions.checkState(!shutdown, "client was shutdown.");
+
+    int numChannels = addresses.size() * channelsPerServer;
+    int numThreads =  numChannels * stubsPerChannel;
+    threadpool = MoreExecutors.listeningDecorator(newFixedThreadPool(numThreads));
+    try {
+      int server_idx = -1;
+      for (InetSocketAddress address : addresses) {
+        server_idx++;
+        for (int i = 0; i < channelsPerServer; i++) {
+          ManagedChannel channel = createChannel(address);
+          channels.add(channel);
+          for (int j = 0; j < stubsPerChannel; j++) {
+            String gaugeName =
+                String.format("/stress_test/server_%d/channel_%d/stub_%d/qps", server_idx, i, j);
+            Worker worker =
+                new Worker(channel, testCaseWeightPairs, durationSecs, gaugeName);
+
+            workerFutures.add(threadpool.submit(worker));
+          }
+        }
+      }
+    } catch (Exception e) {
+      log.log(Level.WARNING, "The stress test client encountered an error!", e);
```

Any reason to not just let the exception bubble up?  It's in the throws declaration of the method sig.